docs/proposal/1/design.md

```diff
@@ -61,61 +61,65 @@ A lightweight **ServiceContainer** will provide access to cross-cutting concerns
 
 ### 7. Data Persistence & State Transfer
 
-To avoid the pitfalls of shared mutable global state and ensure data integrity, we will adopt a **Hydration/Dehydration** model for persistence.
+To avoid the pitfalls of shared mutable global state while solving the performance and boilerplate issues of strict DTOs, we will adopt a **Manifest & Protocol** model.
 
 #### 7.1. Core Principles
 
-- **No Global Session in ECS**: Systems inside a Scene **never** access a global `GameSession`, `SaveFile`, or `Database` object directly. They operate exclusively on local Components.
-- **Explicit Data Transfer**: Data moving between the permanent storage (Session) and the active game (Scene) is always explicit, typed, and snapshotted.
-- **Source of Truth**:
-    - During a Scene's lifecycle, the **ECS World is the single source of truth**.
-    - The persistent storage is updated only during transition boundaries or explicit checkpoints, preventing desync bugs.
+- **No Global Session in ECS**: Systems inside a Scene **never** access a global `GameSession` directly.
+- **MsgSpec for Serialization**: Instead of writing manual `to_dict` methods, Components that require persistence should inherit from `msgspec.Struct`. This provides high-performance, validation-ready serialization out of the box.
+- **Explicit Contracts (Manifests)**: Scenes must declare exactly what data they require from the session. This prevents the "God Object" problem where a SceneManager has to know the internals of every Scene.
 
 #### 7.2. The Lifecycle
 
-1.  **Hydration (Input)**
-    - When the `SceneManager` pushes a new Scene, it extracts necessary data from the Master Save State.
-    - It creates a **SceneContextDTO** (Data Transfer Object). This is a read-only, immutable structure (e.g., a frozen Python dataclass).
+1.  **Scene Manifest (Contract)**
+    - Each Scene class defines a static `Manifest` describing its data dependencies.
     - **Example**:
       ```python
-      @dataclass(frozen=True)
-      class GameplayContext:
-          player_stats: PlayerStatsDTO
-          inventory: List[ItemDTO]
-          active_quests: List[QuestID]
+      class DungeonScene(Scene):
+          MANIFEST = {
+              "required": ["player_inventory", "player_stats"],
```

Comment:
Instead of string could direct class identifier be used or is this not technically possible.

I would like to prevent stringly type data if possible.